ray-curator/api-design.md

```diff
@@ -0,0 +1,196 @@
+# NeMo Curator Ray API Design
```

Comment:
Maybe rename the file to `ARCHITECTURE.md` or should there be another one?

Reply:
Hmmm I'm not fixed on name but I think we can keep it that way. For `ARCHITECTURE.md` I assume it should be more complete than this file. I have intentionally kept this file slim. Eg:- nothing about inputs and outputs.